### PR TITLE
fix: password no longer selected as default

### PR DESCRIPTION
**Pull request by Remakh**

### CODE REVIEW

File: src/services/UserService.ts [hunk 1]
```diff
@@ -105,7 +106,11 @@ export class UserService {
 			throw new APIError(HttpCode.BadRequest, AuthenticateError.AccountNotFound);
 		}
 
-		const user = await getRepository(User).findOne({ email });
+		const user = await getRepository(User)
+			.createQueryBuilder('user').where('user.email= :email', { email })
+			.addSelect('user.password')
+			.getOne();
+
```
Comment (amishshah) [inline src/services/UserService.ts:113]: Could you use this instead? It just means other user relations get loaded by default:

```js
const user = await getRepository(User).findOne({
	where: {
		email
	},
	select: ['password']
});
```

**Review by amishshah — CHANGES_REQUESTED** [review]

File: src/services/UserService.ts (hunk 1 above)
Comment (Remakh) [reply src/services/UserService.ts:113]: Tried it now but that doesn't work.  I've attached a screenshot of the error I received from npm test. 
image

Comment (Remakh) [reply src/services/UserService.ts:113]: Does the query builder not load by default? I assumed it was SELECT * (+ password) WHERE user.email = email?

Comment (seanjparker) [reply src/services/UserService.ts:113]: In Typeorm, if you exclude the password field using `@Column({ select: false })`, and you then want to load it in a query, there as two options iirc:
- Define every column you want to select in the `.findOne()` method call (incl. the password field)
- Use the `QueryBuilder` with `addSelect`

**Review by amishshah — APPROVED** [review]

**Pull request merged**